### ISSUE TITLE
RouteMatched Object

### ISSUE THREAD

**Issue by gianarb** [pennyphp/penny#16]
Create a RouteMatch object to increase ease to change router library

**Comment by gianarb** [comment]
@wdalmut this is an old issue :) Idea?

## Target RouteMatch

The **RouteMatch** job will be: manage interoperability between penny and all other router.. Will be built during `$dispatch->dispatch()` process.

``` php
interface RouteMatch
{
    /**
     * get callable function. This is the action resolved by dispatcher
     */
    public function getCallable();

    /**
     * set callable function.
     */
    public function setCallable(callable $callable);

    /**
     * this is the future event name
     */
    public function setName($name)

     /**
     * this is the future event name
     */
    public function setName($name)
}
```

The problem around this implementation is this caos
https://github.com/pennyphp/penny/blob/master/src/App.php#L114

and this line of code https://github.com/pennyphp/penny/blob/master/src/App.php#L106

I will use "name" to remove this :shit:
https://github.com/pennyphp/penny/blob/master/src/App.php#L106-L111

**Comment by wdalmut** [comment]
Maybe a tiny interface?

``` php
interface RouteMatchInterface
{
    /** @return callable */
    public function getCallable();
    public function getName();
}
```

any class 

``` php
class RouteMatch implements RouteMatchInterface
{
    public function __construct($name, callable $dispatchable) { ... }
    // Implements other methods...
}
```

``` php
function myRouter($request) {
    $matcher = //...
    return $matcher->match($request); // returns a RouteMatchInterface object
}
```

Single line router app

``` php
function router($request)
{
  $matcher = function($request) {
    if ($request->getPath() == "/") {
      return new RouteMatch("homepage", function($request, $response) {
        return $response->withBody("OK");
      });
    }
  };
  return $matcher($request);
}
```

**Comment by wdalmut** [comment]
Or maybe drop the interface and works with a simple array...

``` php
return [
  "name" => "homepage",
  "callable" => function ($request) {
    // ...
  }
];
```

**Comment by gianarb** [comment]
At the moment the problem is that I use DiC to build callable.. :) 
https://github.com/pennyphp/penny/blob/master/src/App.php#L106

The callable is not ready.. I should prepare it

**Comment by gianarb** [comment]
Closed #91
